bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookKeeperTest.java

```diff
@@ -536,12 +536,9 @@ public void testReadAfterLastAddConfirmed() throws Exception {
                 }
             }
 
-            try {
-                writeLh.close();
-                fail("should not be able to close the first LedgerHandler as a recovery has been performed");
-            } catch (BKException.BKMetadataVersionException expected) {
-            }
-
+            // should still be able to close as long as recovery closed the ledger
+            // with the same last entryId and length as in the write handle.
+            writeLh.close();
```

Comment:
This seems like a desirable change to me.  Moreover, it protects against the case where the racing update came from the same client due to ZooKeeperClient resending the update.

Reply:
I understood it is not documented. but if we are changing any existing behavior, I would suggest either doing it in a separate PR, or if it is difficult to do it in a separate PR, then update the javadoc of this method to make things clear "this method will not throw any exceptions anymore if hitting metadata version exceptions"

Reply:
It may still throw an exception on metadata version exception. However, it will only throw the exception if the length or last entry id in the conflicting write is different to what the caller of #close believed it to be. I strongly prefer making this change as part of this patch, as to do otherwise would be to insert arbitrary strange behaviour into the new implementation. I'll add a javadoc for this.

Reply:
added javadoc